source/gui/guiHelper.py

```diff
@@ -140,35 +158,25 @@ def associateElements(firstElement: wx.Control, secondElement: wx.Control) -> wx
 
 	# staticText and input control
 	# likely a labelled control from LabeledControlHelper
-	if isinstance(firstElement, wx.StaticText) and isinstance(
-		secondElement,
-		(
-			wx.Button,
-			wx.Choice,
-			wx.Slider,
-			wx.SpinCtrl,
-			wx.TextCtrl,
-		),
-	):
-		sizer = wx.BoxSizer(wx.HORIZONTAL)
-		sizer.Add(firstElement, flag=wx.ALIGN_CENTER_VERTICAL)
-		sizer.AddSpacer(SPACE_BETWEEN_ASSOCIATED_CONTROL_HORIZONTAL)
-		sizer.Add(secondElement)
-	elif isinstance(firstElement, wx.StaticText) and isinstance(secondElement, wx.CheckBox):
-		# checkbox should go first, and label should go after
-		sizer = wx.BoxSizer(wx.HORIZONTAL)
-		sizer.Add(secondElement)
-		sizer.AddSpacer(SPACE_BETWEEN_ASSOCIATED_CONTROL_HORIZONTAL)
-		sizer.Add(firstElement, flag=wx.ALIGN_CENTER_VERTICAL)
-	# staticText and (ListCtrl, ListBox or TreeCtrl)
-	elif isinstance(firstElement, wx.StaticText) and isinstance(
-		secondElement,
-		(wx.ListCtrl, wx.ListBox, wx.TreeCtrl),
-	):
-		sizer = wx.BoxSizer(wx.VERTICAL)
-		sizer.Add(firstElement)
-		sizer.AddSpacer(SPACE_BETWEEN_ASSOCIATED_CONTROL_VERTICAL)
-		sizer.Add(secondElement, flag=wx.EXPAND, proportion=1)
+	if isinstance(firstElement, wx.StaticText):
+		# Horizontal layout, label first, control second
+		if isinstance(secondElement, _HorizontalCtrlT.__constraints__):
+			sizer = wx.BoxSizer(wx.HORIZONTAL)
+			sizer.Add(firstElement, flag=wx.ALIGN_CENTER_VERTICAL)
+			sizer.AddSpacer(SPACE_BETWEEN_ASSOCIATED_CONTROL_HORIZONTAL)
+			sizer.Add(secondElement)
+		# Horizontal layout, control first, label second
+		elif isinstance(secondElement, wx.CheckBox):
+			sizer = wx.BoxSizer(wx.HORIZONTAL)
+			sizer.Add(secondElement)
+			sizer.AddSpacer(SPACE_BETWEEN_ASSOCIATED_CONTROL_HORIZONTAL)
+			sizer.Add(firstElement, flag=wx.ALIGN_CENTER_VERTICAL)
+		# Vertical layout, label above, control below
+		elif isinstance(secondElement, _VerticalCtrlT.__constraints__):
+			sizer = wx.BoxSizer(wx.VERTICAL)
+			sizer.Add(firstElement)
+			sizer.AddSpacer(SPACE_BETWEEN_ASSOCIATED_CONTROL_VERTICAL)
+			sizer.Add(secondElement, flag=wx.EXPAND, proportion=1)
 	# button and checkBox
 	elif isinstance(firstElement, wx.Button) and isinstance(secondElement, wx.CheckBox):
```

Comment:
Also wonder why this case is in this function.

Inn the config profile dialog, we have the "Triggers..." button associated with the checkbox "Temporarily disable all triggers". IMO, there is no special strong association between these two controls, except the fact that they both apply to triggers. Using `guiHelper.associateElements` instead of a normal sizer / sizer helper is misleading: it let think that the button + the checkbox would act as a specific combined control.
On the opposite, the association of a text box with a button for synth change is really a combined control.

Reply:
again, considering this out of scope. We might want to consider deprecating these inputs in a separate PR?